GA/HeaderFiles/Crossover.h

```diff
@@ -16,137 +19,195 @@ extern int sections;
-					
-					if (intersect_a == false && intersect_b == false)
-					{
-						intersect = false;
-						
-					}
-				}     	 
-				
-				else if (design == "AREA")
-				{
-					// Call constraint AREA for variables
-					bool intersect_A = true;
-					intersect_A = ConstraintAREA(varOutput[i+reproduction_no]);
-					
-					bool intersect_B = true;
-					intersect_B = ConstraintAREA(varOutput[i+1+reproduction_no]);
-					
-					if (intersect_A == false && intersect_B == false)
-					{
-						intersect = false;
-					}
-					
-				}
-				else if (design == "Symmetric Dipole" || design == "Asymmetric Dipole")
-				{
-					// Call constraint Dipole for variables
-					bool intersect_a = true;
-					vector<float> output_a = varOutput[i + reproduction_no][j];
-					intersect_a = ConstraintDipole(output_a[0], output_a[1]);
-
-					bool intersect_b = true;
-					vector<float> output_b = varOutput[i + 1 + reproduction_no][j];
-					intersect_b = ConstraintDipole(output_b[0], output_b[1]);
-
-					if (intersect_a == false && intersect_b == false)
-					{
-						intersect = false;
-
-					}
-				}
-			}
-		}
-		// Save location of the parent antennas
-		selected.push_back(P_loc[locations[i]]);
-		selected.push_back(P_loc[locations[1+i]]);
-	}
-	// Call Mutation to apply mutations on children
-	Mutation(varOutput);
-	
-	// End Flag
-	cout << "Crossover Complete" << endl;
+  // Generate a pair of children from a pair of selected parents
+
+  // Start Flag
+  cout << "Crossover Started" << endl;
+
+  // Define Variables
+  vector<int> locations;
+  vector<int> spare_locations;
+  double exchange;
+  uniform_int_distribution<int> choice(0, 1);
+
+  // call selection methods to populate locations
+  Selection(crossover_no, fitness, locations);
+
+  // call selection to populate spare locations
+  Selection(population, fitness, spare_locations);
+
+  uniform_int_distribution<int> grab(0, spare_locations.size()-1);
+
+  // shuffle vector
+  auto rng = default_random_engine{};
+  shuffle(begin(locations), end(locations), rng);
+  shuffle(begin(spare_locations), end(spare_locations), rng);
+
+
+  // Check the size of location vector
+  if (locations.size() != crossover_no)
+  {
+    cout << "error: parent vector is not proper length" << endl;
+  }
+
+  // Crossover two antennas to make 2 children
+  for (int i = 0; i < crossover_no; i = i + 2)
+  {
+    // Force parents to be different
+    while (dna_input[locations[i]] == dna_input[locations[i + 1]])
+    {
+      int new_parent = grab(generator);
+      swap(locations[i + 1], spare_locations[new_parent]);
+
+    }
+    // If the children are identical to their parent, find new parent
+    bool identical = true;
+    while (identical == true)
+    {
+      for (int j = 0; j < sections; j++)
+      {
+        // If the design self-intersects, find a new design
+        bool intersect = true;
+        while (intersect == true)
```

Comment:
Is there any guarantee that a non self-intersecting design will eventually be found?
In the meeting you mentioned a loop getting stuck -- maybe it's happening here?

You could print some debug messages to show how many loops it does before continuing, and possibly give up after a certain number of loops have been executed without finding a viable solution.  This might also help you understand how much time is spent in this part of the code (especially if you add timestamps), so that you know if it might be worth optimizing these loops.